apps/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,44 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
-import { ListItemComponent } from '../list-item/list-item.component';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
 
 @Component({
   selector: 'app-card',
-  templateUrl: './card.component.html',
-  standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
-})
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
+  template: `
+    <ng-content select="img"></ng-content>
 
-  CardType = CardType;
+    <section>
+      <ng-container *ngFor="let item of list">
+        <ng-template
+          [ngTemplateOutlet]="rowTemplate"
+          [ngTemplateOutletContext]="{ $implicit: item }"></ng-template>
+      </ng-container>
+    </section>
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore
-  ) {}
+    <button
+      class="border border-blue-500 bg-blue-300 p-2 rounded-sm"
+      (click)="add.emit()">
+      Add
+    </button>
+  `,
+  standalone: true,
+  imports: [NgIf, NgFor, NgTemplateOutlet],
+  host: {
+    class: 'border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3',
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CardComponent<T> {
```

Comment:
May I ask how you tell `T` it has an `id` prop? using the `@for` loop now needs a `track` param, so this gives me an error:
![image](https://github.com/tomalaforge/angular-challenges/assets/9698639/4046d024-a64f-4406-a799-2cd31ff363c7)

Reply:
Normally I would tell `T` it expects an `id` prop like this, but then it breaks in the city card component:
```ts
export class CardComponent<T extends {id: string}> {
```
![image](https://github.com/tomalaforge/angular-challenges/assets/9698639/535b5a32-2e56-4a82-97e0-63bf11dd6a4b)

Reply:
Nevermind found my issue, id is a number, so the types were not overlapping...